app/docs/src/reference/actions.md

```diff
@@ -109,6 +110,67 @@ Refresh actions and qualifications are the only functions that can run directly
 on HEAD without going through a change set, since they don't modify resources -
 they only read current state or validate configuration.
 
+## Action Arguments
+
+Action functions take an `Input` argument. It has a `properties` field which
+contains an object that has:
+
+- The `si` properties
+
+  These are the core properties set as meta-data for the function. Name, color,
+  etc.
+
+- The `domain` properties
+
+  These are the properties specified in the schema itself.
+
+- The `resource` data
+
+  This is the output of the last action, stored as the state of the resource. It
+  contains 3 fields:
+
+  - _status_: one of "ok", "warning", or "error"
+  - _message_: an optional message
+  - _payload_: the resource payload itself
+
+- The `resource_value` data
+
+  This is information pulled into the component properties from resource payload
+  data. These are properties added with the `addResourceProp()` method of a
+  components schema.
+
+- Any generated `code`
+
+  Generated code is available as a map, whose key is the name of the code
+  generation function that generated it.
```

Comment:
We should probably mention the "why" here. These exist because the domain 1:1 mapping sometimes doesn't have everything needed to run an action or make data easy digestible via subscription. In the case of Butane, sometimes you have CoreOS butane config that needs to be generated into a format digestible as generic cloud-init contents (e.g. for something like EC2 Instance).

Example flow:

```
domain tree --> generation func that calls out to internet and creates another blob in YAML or JSON or systemd unit or whatever --> action func payload
```